### PR TITLE
Refactor Jkube-kit in order to be independent from maven

### LINKED ISSUE #48: Remove dependencies from jkube-kit-build-api and jkube-kit-build-service-docker out of maven

### PR DESCRIPTION
**Pull request by rohanKanojia**
Related to #30 

Fixes #48 #49

### CODE REVIEW

File: jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/maven/enricher/api/JkubeEnricherContext.java [hunk 1]
```diff
@@ -100,41 +93,26 @@ public GroupArtifactVersion getGav() {
...
-                                                        artifact.getArtifactId(),
-                                                        artifact.getVersion()),
-                               artifact.getType(),
-                               artifact.getScope(),
-                               artifact.getFile()));
-        }
-
-        return dependencies;
+    public List<JkubeProjectDependency> getDependencies(boolean transitive) {
+        return transitive ? getProject().getDependenciesWithTransitive() : getProject().getDependencies();
```
[the remark this answers is not in the document]
Comment (rohanKanojia) [reply jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/maven/enricher/api/JkubeEnricherContext.java]: Umm, [getDependencyArtifacts()](https://maven.apache.org/ref/3.5.2/apidocs/org/apache/maven/project/MavenProject.html#getDependencyArtifacts--) returns all dependencies

[the remark this answers is not in the document]
Comment (rohanKanojia) [reply jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/maven/enricher/api/JkubeEnricherContext.java]: I am using two objects which are set using `getArtifacts()` and `getDependencyArtifacts()`: https://github.com/eclipse/jkube/blob/75cc07b3a796f7e60011e534be94429c84b0cb30/jkube-kit/common-maven/src/main/java/org/eclipse/jkube/kit/common/util/MavenUtil.java#L188

[the remark this answers is not in the document]
Comment (manusa) [reply jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/maven/enricher/api/JkubeEnricherContext.java]: I'm asking about the first case (transitive == true), and just to make sure.

The question is:  does getDependenciesWithTransitive return **all** dependencies too?

I don't understand your answer, you mean that this call gets delegated to > `getArtifacts()`?

[the remark this answers is not in the document]
Comment (rohanKanojia) [reply jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/maven/enricher/api/JkubeEnricherContext.java]: `getDepenenciesWithTransitive()` delegates to `getArtifacts()` (returns all dependencies)
`getDependencies()` delegates to `getDependencyArtifacts()` (returns direct dependencies)

[the remark this answers is not in the document]
Comment (manusa) [reply jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/maven/enricher/api/JkubeEnricherContext.java]: :+1:
